### ISSUE TITLE
Client certificates?

### ISSUE THREAD

**Issue by jbhanks** [zadam/trilium#1577]
Is it possible to set up Trilium to use a client certificate (such as with `ssl_verify_client on;` in nginx? If not, is this a feature that ever might be added?

**Comment by zDEFz** [comment]
You do not need Trilium to use a client certificate when Nginx is already set up with certbot, to use HTTPS.
Example: https://tech.def0.de/

I basically bound my instance to a subdomain, on port 80.

**Comment by jbhanks** [comment]
Yes, a local client certificate, so that only someone with the certificate loaded into their browser can access the page.

**Comment by zadam** [comment]
When the nginx is set up to handle the whole TLS termination (including client auth) then there does not need to be any special support in Trilium ...

**Comment by jbhanks** [comment]
I am confused. Nginx *is* already set up to check client certificates, but in Chrome and Firefox I need to go to settings > security > manage certificates and import the certificate. How does my Trilium client know where to find the certificate on my client machine?
I'm pretty new to this, so apologies if I am being thick.

**Comment by zadam** [comment]
Oh, you mean the destop client and sync. I thought you're using the web version.

This is not supported ATM.

**Comment by jbhanks** [comment]
Ok thanks, it would be a great feature to have. I really don't want any random visitors to even see that I have a Trilium instance.
Do you have any idea what it would take to add that capability? If I understand correctly it just needs to send a specified key file with the http request headers.

**Comment by zadam** [comment]
Yes, it would be useful, but I consider it a rather niche feature.

My advice would be to try to use a proxy server on the client side which accepts unsecured request and attaches a client certificate before proxying.

Looks like nginx can be used in such mode: [https://smallstep.com/hello-mtls/doc/client/nginx-proxy](https://smallstep.com/hello-mtls/doc/client/nginx-proxy)

Or this seems to be a dedicated tool: https://github.com/ghostunnel/ghostunnel

**Comment by 1638410** [comment]
hi there is a new way to use (exactly bypass) mtls

the new mtls verify is act by cloudflate CDN
so, just use hostfile point to the origin vps
